### PR TITLE
Ruler fourier fix

### PR DESCRIPTION
**Pull request by denniscfeng**
New methods added to improve Fourier transform analysis, to improve reliability of detecting ruler tick length

### CODE REVIEW

**Comment by stefanv** [comment]
The tests didn't pass for these changes, so let's rebase on top of Theo's merged changes, and update the test suite.  Let's also compare this PR against current master for accuracy.

@theobdt We looked at the Fourier analysis on Friday and concluded that it conflated two things: the magnitude of the signal (e.g., "long" marks on the ruler have a higher amplitude, when the image is summed vertically), and the match based purely on the frequency.  We addressed that by turning the input into a signal of only ones and zeros.  The resulting spectrum seemed to be much easier to interpret.

File: pipeline.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def main():
         for image_name in image_names:
             if not image_name.lower().endswith(('.png', '.jpg', '.jpeg')):
                 continue
-            image_path = os.path.join(raw_image_path, image_name)
+            image_path = os.path.join(raw_image_path, image_name).replace("\\","/")
```
Comment (stefanv) [inline pipeline.py]: What is happening here?

**Comment by stefanv** [comment]
I don't see the modifications we made to the FFT calculation itself; are those included?

File: pipeline.py (hunk 1 above)
Comment (denniscfeng) [reply pipeline.py]: Previously the image path was joined by doing `raw_image_path + '/' + image_name`, with this new method, however, the os.path module joins the two paths with a `'\'` instead, which causes problems. So we make sure to replace those.

**Comment by denniscfeng** [comment]
They should be, in line 168 in `ruler_detection.py`. They are outside the `fourier` function, though. Maybe we should move those inside there?

File: pipeline.py (hunk 1 above)
Comment (stefanv) [reply pipeline.py]: os.path joins with the operating system separator, which is `/` on unix and `\\` on Windows, so no replacement should be necessary.

**Pull request merged**